targets/simple_switch/main.cpp

```diff
@@ -20,15 +20,59 @@
 
 /* Switch instance */
 
+#include <dlfcn.h>
 #include <bm/SimpleSwitch.h>
 #include <bm/bm_runtime/bm_runtime.h>
 #include <bm/bm_sim/target_parser.h>
 
+#include <sstream>
+#include <string>
+#include <vector>
+
 #include "simple_switch.h"
 
 namespace {
 SimpleSwitch *simple_switch;
-bm::TargetParserBasic *simple_switch_parser;
+
+std::string load_modules_option = "load-modules";
+
+class TargetParserModules : public bm::TargetParserBasic {
+ public:
+  TargetParserModules() {
+    add_string_option(load_modules_option,
+                      "load the given .so files as modules");
+  }
+
+  int parse(const std::vector<std::string> &more_options,
+                    std::ostream *errstream) override {
+    int result = ::bm::TargetParserBasic::parse(more_options, errstream);
+    load_modules();
+    return result;
+  }
+
+ protected:
+  virtual int load_modules() {
```

Comment:
I may be missing something but why is this `virtual`?

Reply:
In case anyone wants to override this class. If you prefer, I can remove it, and if anyone needs it, they can add it later.